dds/DCPS/DataWriterImpl_T.h

```diff
@@ -321,12 +321,12 @@ DataWriterImpl_T
   DataAllocator* data_allocator() const
   {
     return data_allocator_.get();
-  };
+  }
 
   DDS::ReturnCode_t setup_serialization()
   {
     const DDS::DataRepresentationIdSeq repIds =
-      get_effective_data_rep_qos(qos_.representation.value, false);
+      get_writer_effective_data_rep_qos(qos_.representation.value, cdr_encapsulation());
     if (qos_.representation.value.length() > 0) {
       // If the QoS explicitly sets XCDR, XCDR2, or XML, force encapsulation
       cdr_encapsulation(true);
```

Comment:
This looks like it needs to be updated to account for the new UNALIGNED_CDR_DATA_REPRESENTATION

Reply:
Users cannot explicitly set UNALIGNED_CDR_DATA_REPRESENTATION because it is not considered a valid QoS policy in `Qos_Helper::valid`

Reply:
That needs to be changed too